rackcity/views/site_views.py

```diff
@@ -51,175 +64,205 @@ def datacenter_create(request):
+                + "Cannot delete datacenter that still contains racks"
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    assets_in_storage = Asset.objects.filter(offline_storage_site=id)
+    if len(assets_in_storage) > 0:
+        return JsonResponse(
+            {
+                "failure_message": Status.DELETE_ERROR.value
+                + "Cannot delete offline storage site that still contains assets"
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
     try:
-        existing_dc = Site.objects.get(id=id)
+        existing_site = Site.objects.get(id=id)
     except ObjectDoesNotExist:
         return JsonResponse(
             {
-                "failure_message": Status.MODIFY_ERROR.value
-                + "Datacenter"
+                "failure_message": Status.DELETE_ERROR.value
+                + "Site"
                 + GenericFailure.DOES_NOT_EXIST.value,
-                "errors": "No existing datacenter with id=" + str(id),
+                "errors": "No existing site with id=" + str(id),
             },
             status=HTTPStatus.BAD_REQUEST,
         )
-    for field in data.keys():
-        if field != "id":
-            if len(data[field]) == 0:
-                return JsonResponse(
-                    {
-                        "failure_message": Status.MODIFY_ERROR.value
-                        + "field "
-                        + field
-                        + " cannot be empty"
-                    },
-                    status=HTTPStatus.BAD_REQUEST,
-                )
-            if field == "name":
-                dc_with_name = Site.objects.filter(name__iexact=data[field])
-            else:
-                dc_with_name = Site.objects.filter(
-                    abbreviation__iexact=data[field]
-                )
-            if len(dc_with_name) > 0 and dc_with_name[0].id != id:
-                return JsonResponse(
-                    {
-                        "failure_message": Status.MODIFY_ERROR.value
-                        + "Datacenter with name '"
-                        + data[field].lower()
-                        + "' already exists"
-                    },
-                    status=HTTPStatus.BAD_REQUEST,
-                )
-            setattr(existing_dc, field, data[field])
     try:
-        existing_dc.save()
+        existing_site.delete()
     except Exception as error:
         return JsonResponse(
             {
-                "failure_message": Status.MODIFY_ERROR.value
-                + "Datacenter"
-                + GenericFailure.ON_SAVE.value,
+                "failure_message": Status.DELETE_ERROR.value
+                + "Site"
+                + GenericFailure.ON_DELETE.value,
                 "errors": str(error),
             },
             status=HTTPStatus.BAD_REQUEST,
         )
-    log_action(request.user, existing_dc, Action.MODIFY)
+    log_delete(request.user, ElementType.SITE, existing_site.abbreviation)
```

Comment:
`ElementType.SITE` -> `ElementType.SITE.value`!

Reply:
Wanna check that this doesn't happen elsewhere

Reply:
I was wrong but you're crazy

Reply:
https://www.youtube.com/watch?v=k85mRPqvMbE